gnucash/import-export/aqb/dialog-ab-select-imexporter.c

```diff
@@ -0,0 +1,257 @@
+/*
+ * dialog-ab-select-imexporter.c --
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, contact:
+ *
+ * Free Software Foundation           Voice:  +1-617-542-5942
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org
+ */
+
+/**
+ * @internal
+ * @file dialog-ab-select-imexporter.h
+ * @brief Dialog to select AQBanking importer/exporter and format profile.
+ * @author  Copyright (C) 2022 John Ralls <jralls@ceridwen.us>
+ */
+
+#include <config.h>
+
+#include <glib/gi18n.h>
+#include "dialog-ab-select-imexporter.h"
+#include <dialog-utils.h>
+
+static QofLogModule log_module = G_LOG_DOMAIN;
+
+struct _GncABSelectImExDlg
+{
+    GtkWidget *dialog;
+    GtkWidget *parent;
+    GtkListStore *imexporter_list;
+    GtkListStore *profile_list;
+    GtkWidget *select_imexporter;
+    GtkWidget *select_profile;
+    GtkWidget *ok_button;
+
+    AB_BANKING* abi;
+};
+
+// Expose the selection handlers to GtkBuilder.
+G_MODULE_EXPORT gboolean imexporter_changed(GtkTreeSelection* sel,
+                                            gpointer data);
+G_MODULE_EXPORT gboolean profile_changed(GtkTreeSelection* sel, gpointer data);
```

Comment:
It struck me you use G_MODULE_EXPORT here. However looking at our code I see it's used in several locations of the aqbanking code and also in the xml and dbi backends.

Other parts of the code that also work with GtkBuilder don't use this. I also remember a long time ago I added this to assistant-hierarchy.c and had to remove it because it caused issues on Windows.

I had to dig for it but eventually found this explanation in our Changelog.2011:
```
2011-04-23 18:58  cstim
          
        * [r20593] src/gnome-utils/dialog-account.c,
          src/gnome/assistant-hierarchy.c: Win32: Undefine G_MODULE_EXPORT
          in our internal libraries. Please do not use it. The rest of
          gnucash relies on *everything* being exported. This is done by
          mingw's linker, but only if no symbol at all was marked as being
          exported. As soon as at least one symbol is manually marked as
          export, nothing except those marked symbols are exported, which
          will immediately lead to "unresolved symbols" in the win32 build.
```

I have no idea if the same issue happens with MingW64. And also gnucash' modularity has changed a lot since then. Only very few bits are loaded on demand at runtime still: the xml/db backends, and three gnc_modules (aqbanking, ofx and python).

So my question here really is whether we really need it and/or really want it ? As most of the code lives happily without, we can certainly alter the code and/or build system to not require this. But perhaps there are advantages to a controlled exposure of functions. I have no idea at this point.

Reply:
I copied that over from dialog-ab-trans.c because I was having trouble getting the gtkbuilder signal connecting to work. It didn't help and I switched to connecting the signal in code, but didn't remove the `G_MODULE_EXPORT` decl. Of course changing it to static also works because it's completely internal now. That should also be true of dialog-ab-trans.c as it also explicitly connects the signals in code instead of with gtkbuilder.

On the other hand, the uses in the backends date back to Andreas Köhler and Phil Longstaff in 2006 and 2008 respectively while the ones in dialog-ab-trans.c are from a patched that Christian merged a few months *after* writing that commit. All were compiling cleanly before the switch to MSYS2 and of course still do so I'm not entirely sure that Christian's commit comment is correct.

Reply:
I've replaced `G_MODULE_EXPORT` with `static` in both files in aqb. I'll try the backends later, that's obviously out of scope for this PR.

Reply:
Good.

As to Christian's commit comment. I think it was factual. The fact that it wasn't an issue in the places where G_MODULE_EXPORT was used, was because other parts of gnucash effectively only used these exported symbols for those modules. So no issues there. But at the time of his commit I had accidentally introduced use of G_MODULE_EXPORT in parts of gnucash that were never written with that in mind. And suddenly other functions in the same source files as I had added the macros were no longer visible, but still called - causing linker errors on Windows. Those had to be removed again.

Reply:
OK, I guess. Maybe that's changed, because having the symbols in AQB didn't affect the Windows linkage of libgncmod-aqbanking.dll. After all, the symbols with declspecs didn't have external usage at all, they were all callbacks connected to signals in the same files, but there were legitimately public symbols that linked correctly and weren't marked with declspecs.

In the case of the two backends, the marked symbols are the only ones used publicly, though there are unmarked aliases intended for initializing the backends if we were to statically link them.